pkg/authn/k8schain/k8schain_test.go

```diff
@@ -102,7 +102,7 @@ func TestAttachedServiceAccount(t *testing.T) {
 	if err != nil {
 		t.Errorf("Authorization() = %v", err)
 	}
-	if got != want {
+	if *got != *want {
```

Comment:
This will panic if either is nil, which might happen in the error case. Either fatal on error, or I think `reflecf.DeepEqual` might handle the nil case for you.